llvm/lib/Analysis/ScopedNoAliasAA.cpp

```diff
@@ -114,17 +114,26 @@ static void collectMDInDomain(const MDNode *List, const MDNode *Domain,
         Nodes.insert(MD);
 }
 
+/// Collect the set of scoped domains relevant to the noalias scopes.
+void ScopedNoAliasAAResult::collectScopedDomains(
+    const MDNode *NoAlias, SmallPtrSetImpl<const MDNode *> &Domains) const {
+  if (!NoAlias)
+    return;
```

Comment:
I wouldn't include this null check. Either make this a reference, or instead have the instruction as the argument to query the scope metadata